### PR TITLE
A change to bwa aln alignment command to bwa mem. Newer algorithm wit…

### PR DESCRIPTION
**Pull request by vasilislenis**
…hout the need to generate .sai files

### CODE REVIEW

**Comment by ErinBecker** [comment]
Thanks @vasilislenis for putting in this PR and sorry it has taken so long to respond. We're working on preparing the lessons for their first official release now and are working on cleaning things up. It looks like this is a good suggestion based on newish changes to bwa. However, I'm not familiar with the software so I'm going to defer to the maintainers on this one.

@fpsom @lexnederbragt @crazyhottommy @jrherr I'm not familiar with bwa. Could one of you please take a look at this and see if we want to merge this change? It would be nice if this makes the process simpler for the learners.

**Comment by crazyhottommy** [comment]
I use bwa, and bwa mem is indeed a more recent algorithm. no intermediate sai files will be generated if uses bwa mem. I had a write up before http://crazyhottommy.blogspot.com/2017/06/bwa-aln-or-bwa-mem-for-short-reads-36bp.html

>The bwa github page says if reads < 70bp, bwa aln should be used. 

In general, bwa mem should work just fine.

Tommy

**Comment by fpsom** [comment]
@crazyhottommy I agree with that. I propose that we keep the current form (i.e. with `bwa aln`), and then have a callout explaining how the same process can be performed with `bwa mem`, with the choice being down to the user/specific problem.

File: lessons/02-variant-calling-workflow.md [hunk 1]
```diff
@@ -149,7 +141,7 @@ $ samtools view -S -b $sam > $bam
 Sort the BAM file
 
 ```bash
-$ samtools sort -f $bam $sorted_bam
+$ samtools sort $bam $sorted_bam
```
Comment (crazyhottommy) [inline lessons/02-variant-calling-workflow.md]: depending on the version of samtools, samtools sort syntax changed a bit. need to make sure which version one is using https://www.biostars.org/p/169745/

Comment (fpsom) [reply lessons/02-variant-calling-workflow.md]: Absolutely true. And this is also related to [issue #27](https://github.com/datacarpentry/wrangling-genomics/issues/27) - we need to have a clear list of tools and versions to be used.

I'm tempted to find / create docker files that would always maintain the correct version, but this is something to discuss.

Comment (vasilislenis) [reply lessons/02-variant-calling-workflow.md]: “-f ” option samtools sort command is used in quite an old version (older than samtools/0.1.19). However, a small paragraph that explains the differences of the flags from version to version and the importance of “-h” manual could be useful in my opinion.
I believe the same for the bwa.

Comment (aschuerch) [reply lessons/02-variant-calling-workflow.md]: @fpsom, regarding docker: see also datacarpentry/genomics-workshop#28

**Comment by fpsom** [comment]
The content of this episode should now match the content of the `gh-page` branch. However, the file should be renamed to `02-automating_a_workflow.md` as the content of the original PR was to a different filename.

@ErinBecker @crazyhottommy @lexnederbragt @aschuerch @jrherr can you please check that everything is ok? If so, renaming the file should make the PR "mergable". Alternatively, this PR can be closed and all edits moved to the file currently in the `gh-pages`. Open to and willing to do either.

**Pull request merged**